kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/KyuubiRestClient.java

```diff
@@ -0,0 +1,189 @@
+    String baseUrl = String.format("%s/%s", hostUrl, builder.version.getApiNamespace());
+
+    CloseableHttpClient httpclient = initHttpClient(builder);
+
+    String header = initAuthHeader(builder);
+
+    this.httpClient = new RestClient(baseUrl, httpclient, header);
+  }
+
+  public RestClient getHttpClient() {
+    return httpClient;
+  }
+
+  private CloseableHttpClient initHttpClient(Builder builder) {
+    RequestConfig requestConfig =
+        RequestConfig.custom()
+            .setSocketTimeout(builder.socketTimeout)
+            .setConnectTimeout(builder.connectTimeout)
+            .build();
+    SSLConnectionSocketFactory sslSocketFactory;
+    try {
+      TrustStrategy acceptingTrustStrategy = (cert, authType) -> true;
+      SSLContext sslContext =
+          SSLContexts.custom().loadTrustMaterial(null, acceptingTrustStrategy).build();
+      sslSocketFactory = new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE);
+    } catch (Exception e) {
+      LOG.error("Error: ", e);
+      throw new RuntimeException(e);
+    }
+
+    CloseableHttpClient httpclient =
+        HttpClientBuilder.create()
+            .setDefaultRequestConfig(requestConfig)
+            .setSSLSocketFactory(sslSocketFactory)
+            .build();
+    return httpclient;
+  }
+
+  private String initAuthHeader(Builder builder) {
+    String header = "";
+    switch (builder.authSchema) {
+      case BASIC:
+        // no need to set auth header when username is empty
+        if (StringUtils.isNotBlank(builder.username)) {
+          String authorization = String.format("%s:%s", builder.username, builder.password);
+          header =
+              String.format(
+                  "BASIC %s", Base64.getEncoder().encodeToString(authorization.getBytes()));
+        }
+        break;
+      case SPNEGO:
+        try {
+          String server =
+              StringUtils.isBlank(builder.server)
+                  ? new URI(builder.hostUrl).getHost()
+                  : builder.server;
+          header = String.format("NEGOTIATE %s", AuthUtil.generateToken(server));
+        } catch (Exception e) {
+          LOG.error("Error: ", e);
+          throw new RuntimeException(
+              "Failed to generate spnego auth header for " + builder.hostUrl);
+        }
+        break;
+      default:
+        throw new RuntimeException("Unsupported auth schema");
+    }
+    return header;
+  }
+
+  public static class Builder {
+
+    private String hostUrl;
+
+    private String server;
+
+    private ApiVersion version = ApiVersion.V1;
+
+    private AuthSchema authSchema = AuthSchema.BASIC;
+
+    private String username;
+
+    private String password;
+
+    private int socketTimeout = 3000;
+
+    private int connectTimeout = 3000;
+
+    public Builder(String hostUrl) {
+      this.hostUrl = hostUrl;
+    }
+
+    public Builder server(String server) {
+      this.server = server;
```

Comment:
what does server mean. I looked at test code, it seems the server host of SPNEGO isn't it ?

Reply:
I have had the discussion with @turboFei  before.
The field name `server` may be unclear....Need to rename it.

Reply:
spnegoHost?